configure.ac

```
@@ -590,6 +590,14 @@ then
 	darwin*) MACHDEP="darwin";;
 	'')	MACHDEP="unknown";;
     esac
+
+    if test "$ac_sys_system" = "SunOS"; then
+	# For Solaris, there isn't an OS version specific macro defined
+	# in most compilers, so we define one here.
+	SUNOS_VERSION=`echo $ac_sys_release | tr -d '.'`
```

I suggest:

```suggestion
	SUNOS_VERSION=`echo $ac_sys_release | sed -e 's!\.\([0-9]\)$!.0\1!g' | tr -d '.'`
```

I'm not sure about how you build the number:

```
vstinner@mona$ echo 5.1 | tr -d '.'
51
vstinner@mona$ echo 5.10 | tr -d '.'
510
```

I suggest to make sure that you have 2 digits after the dot before removing the dot:

```
vstinner@mona$ echo 5.1 | sed -e 's!\.\([0-9]\)$!.0\1!g' | tr -d '.'
501
vstinner@mona$ echo 5.10 | sed -e 's!\.\([0-9]\)$!.0\1!g' | tr -d '.'
510
```

So Oracle Solaris 5.9 (released in 2002) is formatted as the number 509.

In Python, we had many issues when we switched from Python 3.9 to 3.10, a lot of code was hardcoded to parse the version as 3 characters or compared versions as floatting point number:

```
>>> 3.1 < 3.10
False
```

---

That I didn't consider, that's a good point. It's updated in both files.